controllers/watcher_controller.go

```diff
@@ -1152,6 +1606,7 @@ func (r *WatcherReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&rbacv1.RoleBinding{}).
 		Owns(&batchv1.Job{}).
 		Owns(&corev1.Secret{}).
+		Owns(&routev1.Route{}).
```

Comment:
One thing we are missing is reconfiguration of the route when the certificate/secret for the route is modified. How complex would be to add it?

Reply:
I just tried locally to edit a Watcher instance and change the route secret with the certificate,and the route is changed to use the new certificate after the controller reconciles. Is that what you had in mind? If so I can try to look into adding it into the kuttl tests

Reply:
What I tested is modifying the existing certificate to add/modify the dnsNames, i.e. That recreates the certificate and secret but the route keeps working with the old one.

Reply:
Actually, iiuc that existing certificate modification case is similar to what happens in a certificate renewal situation.

Reply:
I see. Let me try to index that secret in the watcher controller to make sure it watches it, maybe that can work

Reply:
I'd say we need to manage it via hash or something like that but i'm unsure

Reply:
It seems like watching the secret works, I just changed the dnsname in the `Certificate` without modifying the Watcher object and it reconciled and changed the certificate in the route. Tomorrow I'll create a test for it and push the changes

Reply:
@amoralej I've pushed a new version with a kuttl test case called `watcher-tls-certs-change` which patches the certificate and then checks that the certificate in the route (and the services) is the same as one in the secret

Reply:
Thanks, I've also tested it's working fine now.